### ISSUE TITLE
[OverlayTrigger] Does not hide when view is scrolled away

### ISSUE THREAD

**Issue by croraf** [react-bootstrap/react-bootstrap#2919]
react-bootstrap [v0.31.5]
The issue is primarily noticeable on mobile devices (that is touchscreen devices) - I'm using mobile phone with Android and Chrome.

Consider for example this page: http://provjeriplacu.herokuapp.com
To the left there are _info_ glyphs that show tooltip when hovered.

They are implemented using OverlayTrigger component.

Checking the compnent https://github.com/react-bootstrap/react-bootstrap/blob/master/src/OverlayTrigger.js, by default, the overlay is shown when hovered and focused.

On mobile device the overlay gets shown when touched. It gets hidden when touched away.

The issue arises when you open the overlay and then scroll the container containing the Info glyph.
The simplest expected outcome will be that the overlay hides itself, while in OverlayTrigger it stays open at the fixed place meaning it will move relative to the target and look strange.

A bit harder it can be reproduced on PC hovering the mouse on the Info glyph without moving it, and then scroll with the mouse scroll.

Strange behavior described above is shown on the following pictures:
![image](https://user-images.githubusercontent.com/19698636/34482733-2ed3533a-efba-11e7-91d6-a6b1420c5718.png)

![image](https://user-images.githubusercontent.com/19698636/34482746-44c68360-efba-11e7-867c-7b3997e1deb9.png)

**Comment by taion** [comment]
You need to set the container in that case, so that the overlay can be positioned relative to the correct parent.

**Comment by croraf** [comment]
@taion OK, I see. _container_ prop should be added to: https://react-bootstrap.github.io/components.html#overlays-trigger-props.

Unfortunatelly I ran into some other trouble when using container, so I decided I won't use it and will hide the `Overlay` on container scroll instead (using a hack of `OverlayTrigger`) . I see https://github.com/react-bootstrap/react-bootstrap/blob/master/src/OverlayTrigger.js has `state ` variable show which is toggled to show and hide `Overlay`.

I extended `OverlayTrigger `:

```
class MyOverlayTrigger extends OverlayTrigger {

    componentDidMount() {
        super.componentDidMount();

        const scrollPanel = document.getElementsByClassName('panel-body')[0];
        console.log(scrollPanel);

        scrollPanel.onscroll = () => {
            console.log('scrolling');
            console.log(this.hide);
            this.setState({show: false});
        };
    }

    render () {
        console.log('show state:', this.state.show);

        return super.render();
    }
}
```

The render method here `console.log`s correct `this.state.show` so the state is updated (hidden) when I scroll the container. But this is not reflected in the UI, and React plugin does not update the change of `this.state.show` variable and it stays shown.

![image](https://user-images.githubusercontent.com/19698636/34520832-e91ad562-f08a-11e7-8808-4005514ac02e.png)
